clients/client-python/gravitino/filesystem/gvfs.py

```diff
@@ -0,0 +1,574 @@
+        :param identifier: The fileset identifier
+        :param fileset: The fileset
+        :param fs: The file system corresponding to the fileset storage location
+        :param storage_type: The storage type of the fileset storage location
+        :param virtual_path: The virtual fileset path
+        :return The actual path.
+        """
+        virtual_location = self._get_virtual_location(identifier)
+        storage_location = fileset.storage_location()
+        if self._check_mount_single_file(fileset, fs, storage_type):
+            if virtual_path != virtual_location:
+                raise RuntimeError(
+                    f"Path: {virtual_path} should be same with the virtual location: {virtual_location}"
+                    " when the fileset only mounts a single file."
+                )
+            return storage_location
+        return virtual_path.replace(virtual_location, storage_location, 1)
+
+    @staticmethod
+    def _get_virtual_location(identifier: NameIdentifier):
+        """Get the virtual location of the fileset.
+        :param identifier: The name identifier of the fileset
+        :return The virtual location.
+        """
+        return (
+            f"fileset/{identifier.namespace().level(1)}"
+            f"/{identifier.namespace().level(2)}"
+            f"/{identifier.name()}"
+        )
+
+    def _check_mount_single_file(
+        self, fileset: Fileset, fs: AbstractFileSystem, storage_type: StorageType
+    ):
+        """Check if the fileset is mounted a single file.
+        :param fileset: The fileset
+        :param fs: The file system corresponding to the fileset storage location
+        :param storage_type: The storage type of the fileset storage location
+        :return True the fileset is mounted a single file.
+        """
+        result: Dict = fs.info(
+            self._strip_storage_protocol(storage_type, fileset.storage_location())
+        )
+        return result["type"] == "file"
+
+    @staticmethod
+    def _pre_process_path(path):
+        """Pre-process the path.
+         We will uniformly process `gvfs://fileset/{catalog}/{schema}/{fileset_name}/xxx`
+         into the format of `fileset/{catalog}/{schema}/{fileset_name}/xxx`.
+         This is because some implementations of `PyArrow` and `fsspec` can only recognize this format.
+        :param path: The virtual path
+        :return The pre-processed path
+        """
+        if isinstance(path, PurePosixPath):
+            path = path.as_posix()
+        gvfs_prefix = f"{PROTOCOL_NAME}://"
+        if path.startswith(gvfs_prefix):
+            path = path[len(gvfs_prefix) :]
+        if not path.startswith("fileset/"):
+            raise RuntimeError(
+                f"Invalid path:`{path}`. Expected path to start with `fileset/`."
+                " Example: fileset/{fileset_catalog}/{schema}/{fileset_name}/{sub_path}."
+            )
+        return path
```

Comment:
What if the input `path` doesn't include gvfs prefix, do we support such input path?  Also I would suggest to use another variable instead of overwriting `path` to avoid potential issues.

Reply:
According to my tests on some engines based on fsspec implementation, they only support the format of `fileset/fileset_catalog/schema/fileset_name/xxx`(such as llama_index). Therefore, I think it is reasonable to uniformly process them in this format here, because they are likely to call fs.ls or fs.info to obtain path information and pass it to gvfs again.